src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineTests.cs

```diff
@@ -275,6 +276,65 @@ public static NumberValue Execute(NumberValue x, NumberValue y)
             }
         }
 
+        [Fact]
+        public void CustomValueFunction()

```

Comment:
>CustomValueFunction [](http://example.com/codeflow?start=20&length=19)

This example is very close to the "Host object" scenario in: [https://msazure.visualstudio.com/OneAgile/_queries/edit/12955629/?triage=true](https://msazure.visualstudio.com/OneAgile/_queries/edit/12955629/?triage=true)

Reply:
Right, we actually have something already to support these tagged opaque values. If we create fields of type DType.NamedValue, they'll have a tag associated with them, and don't interact with the rest of the system. Functions that expect that type can validate against the tag, so it can be restricted cleanly as well. 

This isn't currently exposed in PowerFx Interpreter, but it's supported in our type system and was implemented a while back for custom pages.

Reply:
We may be able to solve this scenario already with the UnytpedObject support:
[https://github.com/microsoft/Power-Fx/commit/d53e22b5816e8c2d0722d0f38fd2d42688c62619#diff-3425f96c0ed11735bffc552c807588e38b305797cdbd28459430381533aa8705R78](https://github.com/microsoft/Power-Fx/commit/d53e22b5816e8c2d0722d0f38fd2d42688c62619#diff-3425f96c0ed11735bffc552c807588e38b305797cdbd28459430381533aa8705R78)